guvnor-m2repo-editor/guvnor-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/GuvnorM2Repository.java

```diff
@@ -51,78 +50,78 @@
+        this.artifactRepositoryFactory = factory;
     }
 
     @PostConstruct
     public void init() {
-        preferences.load();
         setM2Repos();
     }
 
     private void setM2Repos() {
-        final String M2_REPO_ROOT = FilenameUtils.separatorsToSystem(preferences.getDefaultM2RepoDir());
 
-        final String meReposDir = System.getProperty("org.guvnor.m2repo.dir");
+        this.repositories.addAll(this.artifactRepositoryFactory.getRepositories());
+        this.pomRepositories.addAll(this.artifactRepositoryFactory.getPomRepositories());
+    }
 
-        if (meReposDir == null || meReposDir.trim().isEmpty()) {
-            M2_REPO_DIR = M2_REPO_ROOT;
-        } else {
-            M2_REPO_DIR = meReposDir.trim();
-        }
+    public String getM2RepositoryDir(String repositoryName) {
+        return this.getM2RepositoryRootDir(repositoryName).replaceAll("/$",
+                                                                      "");
+    }
 
-        final LocalArtifactRepository localRepository = new LocalArtifactRepository("local-m2-repo");
-        final FileSystemArtifactRepository fileSystemRepository = new FileSystemArtifactRepository("guvnor-m2-repo",
-                                                                                                   this.M2_REPO_DIR);
-        final DistributionManagementArtifactRepository distributionManagementArtifactRepository = new DistributionManagementArtifactRepository("distribution-management-repo");
+    public String getM2RepositoryRootDir(String repositoryName) {
 
-        repositories.add(localRepository);
-        repositories.add(fileSystemRepository);
-        repositories.add(distributionManagementArtifactRepository);
+        String defaultName = ArtifactRepositoryFactory.GLOBAL_M2_REPO_NAME;
+        if (repositoryName != null && !repositoryName.isEmpty()) {
+            defaultName = repositoryName;
+        }
 
-        pomRepositories.add(localRepository);
-        pomRepositories.add(fileSystemRepository);
-    }
+        final String name = defaultName;
+
+        ArtifactRepository repository = getArtifactRepository(name);
 
-    public String getM2RepositoryRootDir() {
-        if (!M2_REPO_DIR.endsWith(File.separator)) {
-            return M2_REPO_DIR + File.separator;
+        String rootDir = repository.getRootDir();
+        if (!repository.getRootDir().endsWith(File.separator)) {
+            return rootDir + File.separator;
         } else {
-            return M2_REPO_DIR;
+            return rootDir;
         }
     }
 
-    public String getRepositoryURL() {
-        File file = new File(getM2RepositoryRootDir());
+    private ArtifactRepository getArtifactRepository(String name) {
+        return this.repositories
+                .stream()
+                .filter(artifactRepository -> artifactRepository.getName().equals(name))
+                .findFirst().orElseThrow(() -> new RuntimeException(String.format("Repository %s not found",
```

Comment:
I love streams; don't get me wrong. But what's the significance of there being more than one artifact repository that matches the given name!? Is that guaranteed **never** to occur?